### PR TITLE
Add example of Spline Flow in JAX

### PR DESCRIPTION
**Pull request by nsanghi**
## Description

Added notebook port of JAX Flow trained on MNIST from example given at https://github.com/deepmind/distrax/blob/master/examples/flow.py

### Colab link

https://colab.research.google.com/github/nsanghi/probml-notebooks/blob/add_mnist_spline_flow_jax/notebooks/flow_spline_mnist_jax.ipynb 

### Issue 

Issue 
[make some JAX normalizing flows demos #672](https://github.com/probml/pyprobml/issues/672)

### Steps

- [X] Updated the code from distrax example folder to notebook following the PEP 8 guidelines.
- [X] Followed the [CONTRIBUTING.md](https://github.com/probml/pyprobml/blob/master/CONTRIBUTING.md) guidelines.

### Figures

https://gist.github.com/nsanghi/ce9e7269187a525659ba51009910f499

## Checklist:

- [X] Performed a self-review of the code
- [X] Tested on Google Colab.

## Potential problems/Important remarks

Even after 100K steps, the sample is kind of ok and not very high quality. I tried with different hyper parameters but it did not improve a lot. So Left the original settings and just trained for 100K steps instead of 10K steps

### CODE REVIEW

**Comment by murphyk** [comment]
This looks great - I will add to the book. However please add pointer to the [original distrax code](https://github.com/deepmind/distrax/blob/master/examples/flow.py) on which this is based.

**Comment by nsanghi** [comment]
Thanks for the feedback. Made the suggested changes.
